### DISCUSSION TITLE
How to apply decodable renaming in the ConfuserEx GUI?

### DISCUSSION THREAD

**Discussion by blakeex** [mkaring/ConfuserEx#328] [Q&A] [answered]
Hi. I am new to ConfuserEx GUI. I want to create a Symbol Map when I obsfucate an exe using the ConfuserEx GUI.

So far I have done the following:
1. Enable `Generate Debug Symbols` at the Project tab
2. At the `<Global settings>` in Modules, I added `true` to rule, `Normal` for Preset & enable `Inherit protections`
3. At my exe in Modules, I added `true` to rule, Preset as `None`, disable `Inherit protections`, & add rename in Protections

Then I go to Protect! tab, & click `Protect!` button. 
However I could not find a Symbol Map generated.

As far as I have researched in the repo's closed issue in [here](https://github.com/mkaring/ConfuserEx/issues/211#issuecomment-687053756), I need to use `decodable` renaming mode.

Can anyone guide me how to apply `decodable` renaming in the ConfuserEx GUI?
If not, how do I get the Symbol Map when I use ConfuserEx GUI?

Thank you.

**Comment by mkaring** [comment, marked as answer]
Hello,

one major limitation of the GUI is that you can't add parameters to the protections. This means that you can't enable the renaming mode directly in the GUI.
That being said, it's very easy to enable the mode directly in the `*.crproj` file. Best way to do it, is to setup the project using the GUI and save it once done. The project file can be opened with any editor, it's just a simple XML file. You'll find the entries for the protections in there, such as:
```xml
<protection id="rename" />
```

That is the place, where you can add the additional parameters. In your case you want to enable the `sequential` rename mode, so you would exchange the protection entry listed above to:
```xml
<protection id="rename">
  <argument name="mode" value="sequential" />
</protection>
```

Once you did that, you can save the file and open it again in the GUI. The GUI will not show the parameter, but it will use it.

You can find more information regarding the modes and possible settings in the [wiki](https://github.com/mkaring/ConfuserEx/wiki/Name-Protection).
